### PR TITLE
[TRAFODION-2881] HA fixes

### PR DESCRIPTION
**Pull request by zcorrea**
Fixed multiple problems in monitor Allgather() socket reconnect logic.
- Separated node down detection logic from communication errors and timeouts
  to better handle multiple failure scenarios
- Better handling network resets
- Additional trace information
- Fixed 'node up' hang in monitor shell due to TmSync race condition

### CODE REVIEW

**Comment by zcorrea** [comment]
Jenkins, retest

**Comment by zcorrea** [comment]
Jenkins, retest

File: core/sqf/monitor/linux/cmsh.cxx [hunk 1]
```diff
@@ -128,31 +168,97 @@ int CCmsh::GetClusterState( PhysicalNodeNameMap_t &physicalNodeMap )
...
+        PhysicalNodeNameMap_t::iterator it;
+        
+        StringList_t::iterator    alit;
+        for ( alit = nodeStateList_.begin(); alit != nodeStateList_.end() ; alit++ )
+        {
+            ParseNodeStatus( *alit, nodeName, nodeState );
+
+            // TEST_POINT and Exclude List : to force state down on node name 
+            const char *downNodeName = getenv( TP001_NODE_DOWN );
+            const char *downNodeList = getenv( TRAF_EXCLUDE_LIST );
```
Comment (trinakrug) [inline core/sqf/monitor/linux/cmsh.cxx]: Is TRAF_EXCLUDE_LIST obsolete?

Comment (zcorrea) [reply core/sqf/monitor/linux/cmsh.cxx]: Yes, it is! Good catch!

Comment (zcorrea) [reply core/sqf/monitor/linux/cmsh.cxx]: Created JIRA (TRAFODION-2907) to clean up the use of TRAF_EXCLUDE_LIST in the monitor code.

**Pull request merged**